### PR TITLE
Fix bug in synchronize_terminal_measurements transformer

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes https://github.com/quantumlib/Cirq/issues/5934

### CODE REVIEW

File: cirq-core/cirq/transformers/synchronize_terminal_measurements.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def find_terminal_measurements(
         seen_control_keys |= protocols.control_keys(moment)
         if not open_qubits:
             break
-    return terminal_measurements
+    return list(set(terminal_measurements))
```
Comment (pavoljuhas) [inline cirq-core/cirq/transformers/synchronize_terminal_measurements.py:58]: Perhaps we should convert the local `terminal_measurement` to a set to remove one list --> set conversion.

Do we care about the returned list order?

If not we can add a comment to the docstring.

**Review by pavoljuhas — CHANGES_REQUESTED** [review]
Good catch!  Suggested small updates.

File: cirq-core/cirq/transformers/synchronize_terminal_measurements.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/transformers/synchronize_terminal_measurements.py]: done

**Review by pavoljuhas — APPROVED** [review]
LGTM

**Pull request merged**